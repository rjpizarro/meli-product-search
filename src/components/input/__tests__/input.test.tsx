import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Input from '../index'

describe('Input', () => {
    it('should render correctly ', () => {
        const tree = renderer
            .create(<Input label="Input label" />)
            .toJSON()
        const { container } = render(<Input label="Input label" />)

        expect(tree).toMatchSnapshot()
        expect(container.querySelector('[placeholder="Input label"]')).toBeTruthy()
    })

    it('should add a component as adornment', () => {
        const { getByTestId } = render(<Input label="Input label" endAdornment={<div data-testid="endAdornment" />} />);
        const tree = renderer
            .create(<Input label="Input label" />)
            .toJSON()

        expect(tree).toMatchSnapshot()
        expect(getByTestId('endAdornment')).toBeInTheDocument()
    })
})

