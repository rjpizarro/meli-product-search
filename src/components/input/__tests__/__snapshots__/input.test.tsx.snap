// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should add a component as adornment 1`] = `
<div
  className="input"
>
  <input
    aria-label="Input label"
    className="input__input-element"
    placeholder="Input label"
  />
</div>
`;

exports[`Input should render correctly  1`] = `
<div
  className="input"
>
  <input
    aria-label="Input label"
    className="input__input-element"
    placeholder="Input label"
  />
</div>
`;
